docs/reference/plugins/developing_custom_plugins.md

```diff
@@ -0,0 +1,54 @@
+# Developing Custom Plugins
+
+Shift Plugins are developed as custom *PySide2* classes inheriting from `QtWidgets.QWidget`.
+
+## Writing a Plugin Class
+
+Shift's native plugins code, together with a template file can be found in the `shift/plugins` folder. These should serve as a reference for implementing your custom logic and integrating it correctly inside Shift.
+
+The Plugin widget class instances will always be parented to the Shift Main Window so you can start looking at the API index for the `shift.ui.widgets.main.ShiftWindow` class to understand how to access objects like: the current board, the current workflow, its operators or workflow variables.
+
+Advanced users can code their own way through the Shift Python API, however it is strongly recomended to wrap all your Qt signal connections in a dedicated method (see the `MainWidget._connectBoardSignals` in the *pluginTemplate.py* example file).
+
+```python
+def _connectBoardSignals(self):
+    """Standard method to connect to the current board signals."""
+
+    # The signals to be connected from the board...e.g.
+    self.parent().getActiveBoard().workflowExecuted.connect(self._updateContent)
+```
+
+This method should be run for all existing and new board tabs for the Plugin to correctly interact with Shift.
+
+## Adding the Plugin to Shift
+
+The first step to add the Plugin to the Shift UI is to create a *json* file containing the information needed by Shift to source all of your custom Plugin classes. This file should be formatted in the following way:
+
+```json
+{
+  "Operators List": {
+    "menu": "Tools",
+    "path": "./operatorsList.py",
+    "class": "OperatorsList"
+  },
+  "Variables Browser": {
+    "menu": "Tools",
+    "path": "./variablesBrowser.py",
+    "class": "VariablesBrowser"
+  }
+}
+```
+
+Each Plugin main entry name will define the name displayed by the plugin menu entry in the top bar menu. Then for each plugin entry will need:
```

Comment:
Can we agree on how to write plugin? In this documentation is written as `plugin` and `Plugin`. We need to define which is the best way to write it.

Reply:
changed everything to Plugin
done

Reply:
Yes, on this file all are addressed, but please, remember to also change the word in the `docs/reference/plugins.md` file, there are some words still using `plugin` instead of `Plugin`. Thanks!!

Reply:
done